process/accountProcessor.go

```diff
@@ -513,6 +513,10 @@ func (ap *AccountProcessor) getObserversForAddress(address string, availability
 		return nil, err
 	}
 
+	if optionalShardID.HasValue {
```

Comment:
no need to compute the shard id if the value is set
However, you only protected against bad usages of this flag at API level. If one uses proxy as a library and directly call these processors, they might end up with inconsistent results. a simple solution would be renaming shardID to forcedShardID (or something similar) in this case

Reply:
refactored